src/server/common_services/auth_server.js

```diff
@@ -660,12 +662,45 @@ function _get_auth_info(account, system, authorized_by, role, extra) {
     return response;
 }
 
+// has_bucket_permission returns true if either the account is the owner of the bucket
+// or the account has explicit permission to access the bucket via the Bucket Policy.
+//
+// NOTE: If the owner account has is denied access as per the policy then the owner
+// also is denied access to the bucket.
 function has_bucket_permission(bucket, account) {
```

Comment:
it seems like we only use has_bucket_permission in lambda triggers which I think we are going to remove soon. Maybe we should just take it off

Reply:
Actually:
- `req.has_bucket_permission` wraps this function.
    - Gets called only in 2 places but as you mentioned in context of lambdas.
- `req.has_s3_bucket_permission` also wraps this function.
    - `req.check_s3_bucket_permission` wraps the above function.
        - Gets called in `load_bucket` in `object_server.js`
        - Gets called in `find_bucket` in `bucket_server.js`
    - Gets called in `list_buckets` in `bucket_server.js`